src/SWP/Bundle/ContentBundle/EventListener/ProcessArticleMediaListener.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\EventListener;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ODM\PHPCR\Document\Generic;
+use SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\ArticleMedia;
+use SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\Image;
+use SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\ImageRendition;
+use SWP\Bundle\ContentBundle\Event\ArticleEvent;
+use SWP\Bundle\ContentBundle\Manager\MediaManagerInterface;
+use SWP\Bundle\ContentBundle\Model\ArticleInterface;
+use SWP\Component\Bridge\Model\ItemInterface;
+use SWP\Component\Bridge\Model\Rendition;
+use SWP\Component\MultiTenancy\PathBuilder\TenantAwarePathBuilderInterface;
+
+class ProcessArticleMediaListener
```

Comment:
what about splitting that class into smaller classes, like MediaProcessor, RenditionProcessor etc?